src/js/components/PodView.js

```diff
@@ -0,0 +1,370 @@
+        })
+        .map(function (container) {
+          let containerSpec = spec.getContainerSpec(container.name);
+          Object.keys(containerSpec.resources).forEach(function (key) {
+            resourcesSummary[key] = containerSpec.resources[key];
+          });
+
+          return {
+            id: container.getId(),
+            name: container.getName(),
+            status: container.getContainerStatus(),
+            address: container.endpoints.reduce(function (components, ep) {
+              components.push(
+                <a className="text-muted"
+                  href={
+                    'http://' +
+                    instance.getAgentAddress() + ':' +
+                    ep.allocatedHostPort
+                  }
+                  target="_blank"
+                  title="Open in a new window">
+                  {':' + ep.allocatedHostPort}
+                </a>
+              );
+
+              return components;
+            }, []),
+            cpus: containerSpec.resources.cpus,
+            mem: containerSpec.resources.mem,
+            updated: container.getLastUpdated(),
+            version: ''
+          };
+        });
+
+      return {
+        id: instance.getId(),
+        name: instance.getName(),
+        address: instance.getAgentAddress(),
+        cpus: resourcesSummary.cpus,
+        mem: resourcesSummary.mem,
+        updated: instance.getLastUpdated(),
+        status: instance.getInstanceStatus(),
+        version: spec.getVersion(),
+        children
+      };
+    });
+
+  }
+
+  getInstanceFilterStatus(instance) {
+    let status = instance.getInstanceStatus();
+    switch (status) {
+      case PodInstanceStatus.STAGED:
+        return 'staged';
+
+      case PodInstanceStatus.HEALTHY:
+      case PodInstanceStatus.UNHEALTHY:
+      case PodInstanceStatus.RUNNING:
+        return 'active';
+
+      case PodInstanceStatus.KILLED:
+        return 'completed';
+
+      default:
+        return '';
+    }
+  }
+
+  renderWithClickHandler(rowOptions, className, content) {
+    return (
+        <div onClick={rowOptions.clickHandler} className={className}>
+          {content}
+        </div>
+      );
+  }
+
+  renderColumn_ID(prop, row, rowOptions = {}) {
+    if (!rowOptions.isParent) {
+      return (
+        <div className="pod-history-instance-id text-overflow">
+          <Link
+            className="emphasize clickable text-overflow"
+            to="services-task-details"
+            params={{id: this.props.pod.getId(), taskID: row.id}}
+            title={row.name}>
+            <CollapsingString string={row.name} />
+          </Link>
+        </div>
+      );
+    }
+
+    let classes = classNames('pod-history-container-id is-expandable', {
+      'is-expanded': rowOptions.isExpanded
+    });
+
+    return this.renderWithClickHandler(rowOptions, classes, (
+      <CollapsingString string={row.id} />
+    ));
+  }
+
+  renderColumn_Address(prop, row, rowOptions = {}) {
```

Comment:
If there is not any magic behind this function name which I do not know, please do not use `_` underscores in function names.